src/main/java/org/testng/internal/BaseInvoker.java

```diff
@@ -50,7 +50,7 @@ protected void runInvokedMethodListeners(
     }
 
     InvokedMethodListenerInvoker invoker =
-        new InvokedMethodListenerInvoker(listenerMethod, testResult, m_testContext);
+        new InvokedMethodListenerInvoker(listenerMethod, testResult, testResult.getTestContext());
```

Comment:
Are you sure `m_testContext` and `testResult.getTestContext()` are always the same?

Reply:
Yep. They will always be the same. Every TestResult object is always bound to the `ITestContext` to which it belongs.

Reply:
I could be nice to assert it: `assert m_testContext == testResult.getTestContext()`